replica/table.cc

```diff
@@ -2732,7 +2732,7 @@ void tablet_storage_group_manager::update_effective_replication_map(const locato
     //  Also serves as a protection for clearing the cache on the new range, although it shouldn't be a
     //  problem as fresh node won't have any data in new range and migration cleanup invalidates the
     //  range being moved away.
-    if (tablet_migrating_in) {
+    if (tablet_migrating_in || old_tablet_count != new_tablet_count) {
```

Comment:
I don't mind merging this as is, but I don't think tablet count change is a good signal. Maybe we had a split and one of the tablet replicas was migrated away. Tomorrow we'll introduce a tablet bound change.

We need to make the condition test what we actually want to trigger a mutation_source refresh, not looks at metrics that are associated with it.

Reply:
And if we do that, we can run refresh_mutation_source only on affected tablet replicas.

Reply:
> I don't mind merging this as is, but I don't think tablet count change is a good signal. Maybe we had a split and one of the tablet replicas was migrated away. Tomorrow we'll introduce a tablet bound change.
> 
> We need to make the condition test what we actually want to trigger a mutation_source refresh, not looks at metrics that are associated with it.

It's a good point, we could implement a procedure if we had any range movement in this replica by comparing old and tablet map.

Reply:
> And if we do that, we can run refresh_mutation_source only on affected tablet replicas.

the mutation source is table wide, since there's one cache per table. this refreshes our tablet_sstable_set which holds one sstable_set for each tablet replica.